Altis_Life.Altis/core/medical/fn_medicMarkers.sqf

```diff
@@ -1,61 +1,53 @@
-    {
-        _name = _x getVariable "name";
-        _down = _x getVariable ["Revive",false];
-        if (!isNil "_name" && !_down) then {
-            _units pushBack _x;
-        };
-    } forEach allDeadMen;
-
-    {
-        if !(_x isEqualTo player) then {
-            _markerss = createMarkerLocal [format ["%1_marker",_x],visiblePosition _x];
-            _markerss setMarkerColorLocal "ColorIndependent";
-            _markerss setMarkerTypeLocal "Mil_dot";
-            _markerss setMarkerTextLocal format ["%1", _x getVariable ["realname",name _x]];
-
-            _markersMedecin pushBack [_markerss,_x];
-        };
-    } forEach _medics;
-
-    //Loop through and create markers.
-    {
-        _marker = createMarkerLocal [format ["%1_dead_marker",_x],visiblePosition _x];
-        _marker setMarkerColorLocal "ColorRed";
-        _marker setMarkerTypeLocal "loc_Hospital";
-        _marker setMarkerTextLocal format ["%1",(_x getVariable ["name","Unknown Player"])];
-        _markers pushBack _marker;
-    } forEach _units;
-
-    while {visibleMap} do {
-        {
-            private ["_unit"];
-            _unit = _x select 1;
-            if (!isNil "_unit" && !isNull _unit) then {
-                (_x select 0) setMarkerPosLocal (visiblePosition _unit);
-            };
-        } forEach _markersMedecin;
-        if (!visibleMap) exitWith {};
-        sleep 0.02;
-    };
-    {deleteMarkerLocal (_x select 0);} forEach _markersMedecin;
-    _markersMedecin = [];
-    _medics = [];
-    waitUntil {!visibleMap};
-    {deleteMarkerLocal _x;} forEach _markers;
-};
+/*
+    File: fn_medicMarkers.sqf
+    Author: Bryan "Tonic" Boardwine
+
+    Description:
+    Marks downed players on the map when it's open.
+*/
+private _deadMarkers = [];
+private _medicMarkers = [];
+
+{
+    if (side _x isEqualTo independent) then {
+        private _marker = createMarkerLocal [format ["%1_marker",_x],getPosATL _x];
+        _marker setMarkerColorLocal "ColorIndependent";
+        _marker setMarkerTypeLocal "Mil_dot";
+        _marker setMarkerTextLocal format ["%1", _x getVariable ["realname",name _x]];
+        _medicMarkers pushBack [_marker,_x];
+    };
+} forEach (playableUnits - [player]);
+
+{
+    private _name = _x getVariable "name";
+    private _down = _x getVariable ["Revive",false];
+    if (!isNil "_name" && !_down) then {
+        private _marker = createMarkerLocal [format ["%1_dead_marker",_x],getPosATL _x];
+        _marker setMarkerColorLocal "ColorRed";
+        _marker setMarkerTypeLocal "loc_Hospital";
+        _marker setMarkerTextLocal format ["%1",_name];
+        _deadMarkers pushBack _marker;
+    };
+} forEach allDeadMen;
+
+while {visibleMap} do {
+    {
+        _x params [
+            ["_mark",""],
+            ["_unit",objNull]
+        ];
+        if !(isNull _unit) then {
+            _mark setMarkerPosLocal (getPosATL _unit);
+        };
+    } forEach _medicMarkers;

```

Comment:
```suggestion
    } forEach (_medicMarkers + _deadMarkers);

```

Reply:
you didn't do anything with it, so this makes it less redundant code. Might even consider only using one variable for everything